File: csharp/Student.cs
using System;

namespace StudentManagementSystem
{
    public class Student : Person
    {
        private string address;
        public string Phone { get; set; }

        public Student(int id, string name, string address, string phone)
            : base(id, name)
        {
            this.address = address;
            Phone = phone;
        }

        public string GetAddress()
        {
            return address;
        }

        public void Enroll(Program program)
        {
            Console.WriteLine($"{GetName()} enrolled in {program.Name}");
        }

        // Overrides Person's abstract method (polymorphism)
        public override void DisplayRole()
        {
            Console.WriteLine($"{GetName()} is a Student");
        }
    }
}

File: cpp/Student.cpp
#include "Student.h"
#include <iostream>

Student::Student(int id, const std::string& name, const std::string& address, const std::string& phone)
    : Person(id, name), address(address), phone(phone) {}

std::string Student::getAddress() const {
    return address;
}

void Student::enroll(const Program& program) const {
    std::cout << getName() << " enrolled in " << program.name << std::endl;
}

void Student::displayRole() const {
    std::cout << getName() << " is a Student" << std::endl;
}
